processor/fanoutconnector.go

```diff
@@ -37,11 +43,29 @@ var _ MetricsProcessor = (*metricsFanOutConnector)(nil)
 // ConsumeMetricsData exports the MetricsData to all consumers wrapped by the current one.
 func (mfc metricsFanOutConnector) ConsumeMetricsData(ctx context.Context, md consumerdata.MetricsData) error {
 	var errs []error
-	for _, mc := range mfc {
-		if err := mc.ConsumeMetricsData(ctx, md); err != nil {
+
+	// Fan out to first len-1 consumers.
+	for i := 0; i < len(mfc)-1; i++ {
+		// Create a clone of data. We need to clone because consumers may modify the data.
+		clone, err := cloneMetricsData(&md)
```

Comment:
`err` will always be nil so it can removed from the func return values, as I mentioned in https://github.com/open-telemetry/opentelemetry-service/pull/367/files#r328816757.

Reply:
👍 

Note to self: this is what happens when you rush too much.